File: LMS/LMS.Application/Filters/HR/EmployeeIdentity/IncentiveFilter.cs
using LMS.Application.Filters.Base;

namespace LMS.Application.Filters.HR.EmployeeIdentity
{
    public class IncentiveFilter : Filter
    {
        public List<Guid>? DepartmentIds { get; set; }
        public bool? ByIsPaid { get; set; }
        public bool? ByIsApproved { get; set; }

        public bool IsDesc { get; set; } = true;
        public IncentiveOrderBy OrderBy { get; set; }
    }
}

File: LMS/LMS.UI/huda-center/src/app/shared/filters/hr/incentive.filter.ts
import { Filter } from '../base/filter';


export enum IncentiveOrderBy {
    ByEmployeeName = 0,
    ByDate = 1,
    ByReason = 2,
    ByIsPaid = 3,
    ByIsApproved = 4
}

export interface IncentiveFilter extends Filter {
    departmentIds?: string[];
    byIsPaid?: boolean | null;
    byIsApproved?: boolean | null;
    isDesc?: boolean;
    orderBy?: IncentiveOrderBy;
}
